src/public_info.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement.  This, along with the Licenses can be
+// found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+// FIXME: remove when this module is finished
+#![allow(dead_code)]
+
+use crust::Uid;
+use rust_sodium::crypto::{box_, sign};
+use std::fmt::{self, Debug, Display, Formatter};
+use xor_name::XorName;
+
+/// Public identifier for a Peer, contains age if peer is a node.
+#[derive(Serialize, Deserialize, PartialOrd, Ord, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum PublicInfo {
+    /// Public identifier for a client
+    Client {
+        /// Client's public signing key
+        sign_key: sign::PublicKey,
```

Comment:
Can we keep these fields in the same order throughout please?  The `FullInfo` (and task spec) has encrypt keys first.

Reply:
There is a test `public_info_order` requires `name` (now the sign_key) to be preferred other than encrypt_key.
I can change the keys order for FullInfo to make them consistent.

Reply:
I don't think that test is particularly useful.  We can't just use the internals of the `PublicInfo` to define the sort order for the case of choosing the oldest node.  We need to accommodate the tie breaker rule.

Either way, I think we should have consistent ordering of the member variables and functions across the two files!